### PR TITLE
HDDS-3151. TestCloseContainerHandlingByClient Enable-testMultiBlockW…

### PR DESCRIPTION
**Pull request by aryangupta1998**
…rites3 and new TestDiscardPreallocatedBlocks

## What changes were proposed in this pull request?

Enabled testMultiBlockWrites3 and removed testDiscardPreallocatedBlocks because of changed configurations and created new test TestDiscardPreallocatedBlocks.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3151

## How was this patch tested?

Tested Manually

### CODE REVIEW

**Review by adoroszlai — COMMENTED** [review]
Thanks @aryangupta1998 for working on this.  Can you please try running these two test classes repeatedly?

https://cwiki.apache.org/confluence/display/HADOOP/Github+Actions+tips+and+tricks#GithubActionstipsandtricks-Executingonetestmultipletimes

**Comment by aryangupta1998** [comment]
> Thanks @aryangupta1998 for working on this. Can you please try running these two test classes repeatedly?
> 
> https://cwiki.apache.org/confluence/display/HADOOP/Github+Actions+tips+and+tricks#GithubActionstipsandtricks-Executingonetestmultipletimes

Thanks, @adoroszlai for the comment. I have created a workflow, test passed for 20 iterations. Kindly check [https://github.com/aryangupta1998/hadoop-ozone/actions/runs/227288635](url)

**Comment by adoroszlai** [comment]
> test passed for 20 iterations

Thanks.  Do you know what caused `testMultiBlockWrites3` failure earlier, and what previous code change makes it possible to simply enable it now?

**Comment by bshashikant** [comment]
> > test passed for 20 iterations
> 
> Thanks. Do you know what caused `testMultiBlockWrites3` failure earlier, and what previous code change makes it possible to simply enable it now?

I think the previous failures were related to retry policy in ratis. As these are addresssed , these tests should be stable now.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestDiscardPreallocatedBlocks.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  @Test
+  public void testDiscardPreallocatedBlocks() throws Exception {
+    String keyName = getKeyName();
+    OzoneOutputStream key =
+        createKey(keyName, ReplicationType.RATIS, 2 * blockSize);
+    KeyOutputStream keyOutputStream =
+        (KeyOutputStream) key.getOutputStream();
+    Assert.assertTrue(key.getOutputStream() instanceof KeyOutputStream);
+    // With the initial size provided, it should have pre allocated 2 blocks
+    Assert.assertEquals(2, keyOutputStream.getStreamEntries().size());
```
Comment (bshashikant) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestDiscardPreallocatedBlocks.java:132]: Let's also add an assertion condition to verify the container Ids for both preallocated blocks are same.

Comment (aryangupta1998) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestDiscardPreallocatedBlocks.java:132]: Added.

**Review by bshashikant — APPROVED** [review]

**Comment by bshashikant** [comment]
Thanks @aryangupta1998 for the contribution.

**Pull request merged**